toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon

```diff
@@ -0,0 +1,253 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/packages/no_prelude/missing_prelude.carbon
+
+// --- fail_missing_prelude.carbon
+
+library "fail_missing_prelude";
+
+// CHECK:STDERR: fail_missing_prelude.carbon:[[@LINE+4]]:8: ERROR: Package `Core` implicitly referenced here, but not found.
+// CHECK:STDERR: var n: i32;
+// CHECK:STDERR:        ^~~
+// CHECK:STDERR:
+var n: i32;
+
+// --- prelude_empty.carbon
+
+package Core library "empty";
+
+// --- fail_missing_prelude_member.carbon
+
+library "fail_missing_prelude_member";
+
+import Core library "empty";
+
+// CHECK:STDERR: fail_missing_prelude_member.carbon:[[@LINE+3]]:8: ERROR: Name `Core.Int32` implicitly referenced here, but not found.
+// CHECK:STDERR: var n: i32;
+// CHECK:STDERR:        ^~~
+var n: i32;
+
+// --- prelude_fake_int.carbon
+
+package Core library "fake_int";
+
+fn Int32() -> {} { return {}; }
+
+// --- use_fake_int.carbon
+
+library "use_fake_int";
+
+import Core library "fake_int";
+
+var n: {} = i32;
+
+// --- prelude_use_in_prelude.carbon
+
+package Core library "prelude_use_in_prelude";
+
+// Core is not an imported package here.
+
+fn Int32() -> () { return (); }
+
+var n: () = i32;
```

Comment:
Other tests use `{}`, so suggesting for consistency.

```suggestion
fn Int32() -> {} { return {}; }

var n: {} = i32;
```

Reply:
I was using a different type here to ensure we were picking the definition from this file, not from the other library.